docs/plugin-development.rst

```diff
@@ -612,72 +609,41 @@ Each type needs to be installed differently due to how node manages external pac
-  ``npm install`` in the plugin directory. This may have certain benefits for
-  plugin development, such as allowing plugin sources to import modules without
-  the alias prefix as described above (though, this alias would still be
-  available for use by other plugins that want to access your plugin's
-  dependencies). Additionally, if your plugin is installed without using
-  symlinks, then you will still have access to Girder's Node dependencies (see
-  this [GitHub conversation](https://github.com/nodejs/node/issues/3402) for a
-  discussion of why symlinked directories will not allow for the usual Node
-  import semantics).
-
-  If instead you are using a custom Grunt build with a Gruntfile, the
-  dependencies should be installed into your plugin's **node_modules** directory
-  by providing a `package.json <https://docs.npmjs.com/files/package.json>`_
-  file just as they are used for standalone node applications.  When such a file
-  exists in your plugin directory, ``npm install`` will be executed in a new
-  process from within your package's directory.
-
-- Build time dependencies that your Grunt tasks rely on to assemble the sources
-  for deployment need to be installed into Girder's own **node_modules** directory.
-  These dependencies will typically be Grunt extensions defining extra tasks used
-  by your build.  Such dependencies should be listed under ``grunt.dependencies``
-  as an object (much like dependencies in **package.json**) inside your
-  **plugin.json** or **plugin.yml** file.
+The ``npm.file`` property is a path to a JSON file relative to the plugin
+directory (``package.json`` is a convenient choice, simply because the ``npm
+install --save-dev`` command manipulates this file by default), while
+``npm.fields`` specifies which top-level keys in that file contain package names
+to install (by default, this property has the value ``['devDependencies',
+'dependencies', 'optionalDependencies']``). If the ``localNodeModules`` option
+is set to ``true``, then the dependencies will be installed within a separate
+directory so that they will not collide with Girder's own set of node_modules.
 
-  .. code-block:: json
+The final alternative for Webpack-built plugins is to set the ``npm.install``
+configuration property to ``true``; this will cause the build system to run
+``npm install`` in the plugin directory.
 
-      {
-          "name": "MY_PLUGIN",
-          "grunt": {
-              "dependencies": {
-                  "grunt-contrib-concat": ">=1.0.1"
-              }
-          }
-      }
+When you use the `import` directive within your plugin code, for example:
+
+.. code-block:: javascript
+
+    import foobar from 'foobar';
+
+The build process will search for the ``'foobar'`` module in the following locations, in order:
+
+1. The plugin's local modules directory that is created if ``npm.localNodeModules`` has
+  been set to ``true`` in the plugin configuration file.
```

Comment:
This still appears just a tad off, at least on GitHub. Note the quote bar on the continuation lines:

![image](https://user-images.githubusercontent.com/81305/30076177-1b82b7a2-9246-11e7-8066-5736141191a5.png)